Source/WebCore/rendering/RenderThemeMac.mm

```diff
@@ -1581,14 +1582,18 @@ static void paintAttachmentPlaceholderBorder(const RenderAttachment& attachment,
 
     bool usePlaceholder = validProgress && !progress;
 
-    paintAttachmentIconBackground(attachment, context, layout);
+    if (!element.isLegacyUnderModern())
```

Comment:
Re-naming `LegacyUnderModern` would also make this code cleaner. This method should really describe the behavior (along the lines of "drawsIconOnly", rather than discuss the "type" of element).